src/InfoBarWidget.js

```diff
@@ -0,0 +1,122 @@
+var InfoBarWidget = function InfoBarWidget( config ) {
+	var widget = this;
+
+	// Configuration initialization
+	config = config || {};
+
+	// Parent constructor
+	OO.ui.ButtonWidget.parent.call( this, config );
+	OO.ui.mixin.IconElement.call( this, config );
+	OO.ui.mixin.LabelElement.call( this, config );
+	OO.ui.mixin.TitledElement.call( this, config );
+	OO.ui.mixin.FlaggedElement.call( this, config );
```

Comment:
couldn't you just do `OO.ui.mixin.IconElement( config )` (same for the other lines)?

Reply:
Unfortunately no, because it's not exactly a function, it's an object constructor. If you call it directly, its context would be itself, and we need the context to be this object; that's why we use `.call` and give it the current context (`this`).

Reply:
You can try this in the console:

```
var ContextTester = function ( name ) { this.name = name; };
ContextTester.prototype.printName = function () { console.log( this.name ); }
var Tester = function () {
   this.name = 'tester';
}
Tester.prototype.test = function () {
   var o = new ContextTester( 'one' );
   o.printName(); // 'one'
   o.printName.call( this ); // 'tester'
};
var t = new Tester();
t.test();
```

Reply:
Ah, that makes sense. Thanks for the explanation!